apps/src/lib/client/signing.rs

```diff
@@ -286,33 +336,715 @@ pub async fn sign_wrapper(
+        epoch,
+        args.gas_limit.clone(),
+        #[cfg(not(feature = "mainnet"))]
+        pow_solution,
+    )));
+    tx.header.chain_id = ctx.config.ledger.chain_id.clone();
+    tx.header.expiration = args.expiration;
+    // Then sign over the bound wrapper
+    tx.add_section(Section::Signature(Signature::new(
+        &tx.header_hash(),
+        keypair,
+    )));
+
+    // Attempt to decode the construction
+    if let Ok(path) = env::var(ENV_VAR_LEDGER_LOG_PATH) {
+        let mut tx = tx.clone();
+        // Contract the large data blobs in the transaction
+        tx.wallet_filter();
+        // Convert the transaction to Ledger format
+        let decoding =
+            to_ledger_vector(ctx, &tx).expect("unable to decode transaction");
+        let output = serde_json::to_string(&decoding)
+            .expect("failed to serialize decoding");
+        // Record the transaction at the identified path
+        let mut f = File::options()
+            .append(true)
+            .create(true)
+            .open(path)
+            .expect("failed to open test vector file");
+        writeln!(f, "{},", output)
+            .expect("unable to write test vector to file");
+    }
+    // Attempt to decode the construction
+    if let Ok(path) = env::var(ENV_VAR_TX_LOG_PATH) {
+        let mut tx = tx.clone();
+        // Contract the large data blobs in the transaction
+        tx.wallet_filter();
+        // Record the transaction at the identified path
+        let mut f = File::options()
+            .append(true)
+            .create(true)
+            .open(path)
+            .expect("failed to open test vector file");
+        writeln!(f, "{:x?},", tx).expect("unable to write test vector to file");
+    }
 
+    // Remove all the sensitive sections
+    tx.protocol_filter();
+    // Encrypt all sections not relating to the header
+    tx.encrypt(&Default::default());
     // We use this to determine when the wrapper tx makes it on-chain
-    let wrapper_hash = hash_tx(&tx.try_to_vec().unwrap()).to_string();
+    let wrapper_hash = tx.header_hash().to_string();
     // We use this to determine when the decrypted inner tx makes it
     // on-chain
-    let decrypted_hash = tx.tx_hash.to_string();
+    let decrypted_hash = tx
+        .clone()
+        .update_header(TxType::Raw)
+        .header_hash()
+        .to_string();
     TxBroadcastData::Wrapper {
-        tx: tx
-            .sign(keypair, ctx.config.ledger.chain_id.clone(), args.expiration)
-            .expect("Wrapper tx signing keypair should be correct"),
+        tx,
         wrapper_hash,
         decrypted_hash,
     }
 }
+
+/// Represents the transaction data that is displayed on a Ledger device
+#[derive(Default, Serialize, Deserialize)]
+struct LedgerVector {
+    blob: String,
+    index: u64,
+    name: String,
+    output: Vec<String>,
+    output_expert: Vec<String>,
+    valid: bool,
+}
+
+/// The tokens that will be hardcoded into the wallet
```

Comment:
let's not add this back in - the token addresses and their aliases can now be looked up from the wallet (https://github.com/anoma/namada/pull/1081)

Reply:
I've updated this in 4d4f7472af1296bf161a02123ab8391c51626999